spec/system/users/edit_spec.rb

```diff
@@ -0,0 +1,48 @@
+RSpec.describe "Editing user" do
+  let(:user) { create(:user, username: 'John Doe', password: 'known', email: 'dummy@example.com') }
+
+  scenario "Logged-out user tries to edit a user" do
+    visit edit_user_path(user)
+    expect(page).to have_selector('.error', text: 'You must be logged in to view that page.')
+    expect(page).to have_current_path(root_path)
+  end
+
+  scenario "User tries to edit a different user" do
+    login
+    visit edit_user_path(user)
+    expect(page).to have_selector('.error', text: 'You do not have permission to modify this account.')
+    expect(page).to have_current_path(continuities_path)
+  end
+
+  scenario "User edits themself", :js do
+    login(user, 'known')
+    visit edit_user_path(user)
+    expect(page).to have_no_selector('.error')
+    expect(page).to have_selector('.editor-title', exact_text: 'Account Settings')
+
+    # TODO all fields
+    # within("#edit_user_#{user.id}") do
+    #   expect(page).to have_field('Username', with: 'John Doe')
+    #   expect(page).to have_field('Email address', with: 'dummy@example.com')
+    #   fill_in 'Username', with: 'Jane Doe'
+    #   fill_in "Email address", with: "dummy2@example.com"
+    # end
+
+    expect(page).to have_no_selector('.select2-selection__choice')
```

Comment:
Wow, you're right, this is horrible :D 

I'll see if we can do some better helpers for this kind of selection logic

Reply:
:)